regression/pubsub.js

```diff
@@ -198,9 +205,40 @@ describe('pubsub', function() {
         assert.equal(msg.data, 'hello');
         subscription.ack(msg.id, done);
       });
-      topic.publishRaw({
-        data: new Buffer('hello').toString('base64')
-      }, assert.ifError);
+      topic.publish([
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') }
+      ], assert.ifError);
+    });
+
+    it.only('should receive the chosen amount of results', function(done) {
+      var subscription = topic.subscription(subscriptions[0].name);
+      var opts = { returnImmediately: true, maxCount: 3 };
+
+      subscription.pull(opts, function(err, messages) {
```

Comment:
If you want, you could set `returnImmediately` to false and then it should wait until messages are available to return.